docs/security/README.md

```diff
@@ -0,0 +1,227 @@
+*OEK* with a new *KEK* derived from the new data key / *EK*:
+
+```
+              object metadata                                         KMS
+                    |                                                  |
+                    |                     +----------------+        1a |  +-------+
+                    |-------------------->| EK_1_encrypted |-----------|->| CMK_1 |
+                    |                     +----------------+           |  +---+---+
+                    |                                                  |      |
+                    |              +---------------+       +------+ 1b |      |
+                    |------------->| OEK_encrypted |       | EK_1 |<---|------+
+                    |              +-------+-------+       +------+    |
+                    |                       \             /            |
+                    |                        \___  2  ___/             |
+                    |                            \___/                 |
+                    |                              |                   |
+                    |                           +--+--+                |
+                    |                           | OEK |                |  +-------+
+                    |                           +--+--+                |  | CMK_2 |
+                    |                              |                   |  +---+---+
+                    |                              |                   |      |
+                    |       +----------------+     |4      +------+ 3a |      |
+                    |<------| OEK_encrypted' |<----+-------| EK_2 |<---|------+
+                    |       +----------------+             +------+    |      |
+                    |                    +----------------+         3b |      |
+                    |<-------------------| EK_2_encrypted |<-----------|------+
+                    |                    +----------------+            |
+                    |                                                  |
+
+
+1a) Send encrypted data key and master key ID to KMS.
+1b) Receive decrypted data key.
+2)  Decrypt encrypted object key with the KEK derived from the data key.
+3a) Receive new plain data key from the KMS using the master key ID of the server config.
+3b) Receive encrypted form of the data key from the KMS.
+4)  Derive a new KEK from the new data key and re-encrypt the OEK with it.
+5)  Store the encrypted OEK, encrypted data key and master key ID in object metadata.
+ ```
+<center>Figure 3 - KMS data key rotation</center>
+
+#### Key rotation - Extensions
+
+The basic SSE-S3 key rotation operation can be used to build more powerful key management operations.
+However, the following options are not implemented or completely designed, yet.
+
+1. **Master key migration**: The [SSE-S3 key rotation](#Key-rotation---Basic-Operation) can be performed
```

Comment:
This section seems ambiguous, we mention these are not implemented or designed, then we use the word "can" which implies the functionality exists.  Might add a qualifier such as "once implemented"?

Reply:
Well, it's possible to do all these actions right now. What I mean here: It's not at all convenient to perform this operations - for example because there may be a Admin-API in the future which provides a nice interface for that.
This section should mainly list some things that will be possible to do with the basic SSE-S3 key rotation operation.

Any idea to express this better?

Reply:
What do you think about "The following options are possible to perform manually but do not have fully functional API's at this time"?